service/topology_coordinator.cc

```diff
@@ -2349,7 +2478,13 @@ future<> topology_coordinator::run() {
                 continue;
             }
 
+            if (_topo_sm._topology.ustate != topology::upgrade_state::done) {
+                co_await do_upgrade_step(std::move(guard));
```

Comment:
So I wonder is it OK to start `cdc_generation_publisher_fiber()` and `start_tablet_load_stats_refresher()` above before upgrade is finished. May be the monitor fiber can run different functions based on `_topo_sm._topology.ustate` state `topology__coordinator::run` vs `topology__coordinator::run_upgrade`?

BTW I would rename `ustate` to `upgrade_state`

Reply:
> So I wonder is it OK to start `cdc_generation_publisher_fiber()` and `start_tablet_load_stats_refresher()` above before upgrade is finished.

Good point, it definitely sounds safer to start them only after upgrade is finished. There is a similar issue with `start_tablet_split_monitor` and `sstable_cleanup_fiber` that are started in `track_upgrade_progress_to_topology_coordinator`, I'll fix them, too.

> May be the monitor fiber can run different functions based on `_topo_sm._topology.ustate` state `topology__coordinator::run` vs `topology__coordinator::run_upgrade`?

Hmm yeah, I could introduce `topology_coordinator::run_upgrade` and call it in `run_topology_coordinator`. I'll try to reuse the error handling logic from `topology_coordinator::run()`.

> BTW I would rename `ustate` to `upgrade_state`

I followed the convention of the existing `tstate`, but makes sense to make `ustate` more descriptive. The only problem is that we already have an enum called `upgrade_state`. I suppose I could rename `upgrade_state` to `upgrade_state_type`?

Reply:
Yes, rename it to _type. tstate is also not that great.